process/block/sovereignChainBlock.go

```diff
@@ -890,40 +887,33 @@ func (scbp *sovereignChainBlockProcessor) processEpochStartMetaBlock(
 		return err
 	}
 
-	// TODO MX-15589 check how we can integrate all of these later on
-	//err = mp.epochRewardsCreator.VerifyRewardsMiniBlocks(header, allValidatorsInfo, computedEconomics)
-	//if err != nil {
-	//	return err
-	//}
-	//
-	//err = mp.epochSystemSCProcessor.ProcessDelegationRewards(body.MiniBlocks, mp.epochRewardsCreator.GetLocalTxCache())
-	//if err != nil {
-	//	return err
-	//}
+	rewardMiniBlocks, err := scbp.epochRewardsCreator.CreateRewardsMiniBlocks(sovHdr, allValidatorsInfo, &sovHdr.EpochStart.Economics)
+	if err != nil {
+		return err
+	}
+
+	sovHdr.EpochStart.Economics.RewardsForProtocolSustainability.Set(scbp.epochRewardsCreator.GetProtocolSustainabilityRewards())
 
-	//err = mp.validatorInfoCreator.VerifyValidatorInfoMiniBlocks(body.MiniBlocks, allValidatorsInfo)
-	//if err != nil {
-	//	return err
-	//}
+	err = scbp.epochSystemSCProcessor.ProcessDelegationRewards(rewardMiniBlocks, scbp.epochRewardsCreator.GetLocalTxCache())
+	if err != nil {
+		return err
+	}
```

Comment:
We need to process the created Rewards TXs as well, like AddToBalance. I did not see that.

Reply:
Do you mean the execution that happens in this func: `func (srtp *sovereignRewardsTxPreProcessor) receivedRewardTx`?

When we create reward txs mbs, we add them to pool. Upon receiving them in the pool, that function is called, which executes reward txs:
```go
	return srtp.rewardsProcessor.ProcessRewardTransaction(rwdTx)
```

Reply:
Or is there anything else that should be executed?

Reply:
Nope. You have to call it here. Maybe we can add it into the epochSystemSCProcessor.